school.py

```diff
@@ -69,26 +43,70 @@ def setup_schools(args, net: sumolib.net.Net, stats: ET.ElementTree, school_coun
-        logging.debug(f"Using begin_age: {begin_age}, end_age: {end_age} for school(s)")
+    logging.info("Inserting " + str(len(new_school_edges)) + f" new {school_type}(s)")
+    for school_edge in new_school_edges:
+        begin_age = random.randint(int(args[f"--{school_type}.begin-age"].split(",")[0]),
+                                   int(args[f"--{school_type}.begin-age"].split(",")[1]))
+        end_age = random.randint(int(args[f"--{school_type}.end-age"].split(",")[0]) if begin_age + 1 <= int(
+            args[f"--{school_type}.end-age"].split(",")[0]) else begin_age + 1,
+                                 int(args[f"--{school_type}.end-age"].split(",")[1]))
+        logging.debug(f"Using begin_age: {begin_age}, end_age: {end_age} for {school_type}(s)")
 
         ET.SubElement(xml_schools, "school", attrib={
-            "edge": str(school.getID()),
-            "pos": str(random.randint(0, 100)),
+            "edge": str(school_edge.getID()),
+            "pos": str(random.randint(0, int(school_edge.getLength()))),
             "beginAge": str(begin_age),
             "endAge": str(end_age),
-            "capacity": str(random.randint(int(args["--schools.capacity"].split(",")[0]),
-                                           int(args["--schools.capacity"].split(",")[1]))),
+            "capacity": str(random.randint(int(args[f"--{school_type}.capacity"].split(",")[0]),
+                                           int(args[f"--{school_type}.capacity"].split(",")[1]))),
             "opening": str(random.randrange(school_open_earliest, school_open_latest, school_stepsize)),
             "closing": str(random.randrange(school_close_earliest, school_close_latest, school_stepsize))
         })
+
+
+def get_school_count(args, stats: ET.ElementTree, school_type: str):
+    if args[f"--{school_type}.count"] == "auto":
+        schools_per_1000_inhabitants = float(args[f"--{school_type}.ratio"])
+
+        # Calculate default number of schools, based on population if none input parameter
+        xml_general = stats.find('general')
+        inhabitants = xml_general.get('inhabitants')
+        school_count = int(inhabitants) * schools_per_1000_inhabitants // 1000
+
+    else:
+        # Else place new number of schools as according to input
+        school_count = int(args[f"--{school_type}.count"])
+
+    return school_count
+
+
+def setup_type_of_school(args, net: sumolib.net.Net, stats: ET.ElementTree, centre: Tuple[float, float], school_type):
+    # Get number of primary schools to be placed
+    school_count = int(get_school_count(args, stats, school_type))
+
+    # Find edges to place schools on
+    if 0 < school_count:
+        new_school_edges = find_school_edges(net, school_count, centre)
+
+        # Insert schools on edges found
+        insert_schools(args, new_school_edges, stats, school_type)
+    else:
+        logging.info(f"Inserting {school_count} new {school_type}(s)")
```

Comment:
The true-branch prints something similar in `insert_schools` (see my other comment). Maybe that should be extracted and put right before this if-statement to avoid duplicate code.

Reply:
Did the opposite in 433d129035ebe74d19701c40291dee1be62ff5f4
Reason in removing the logging in the else branch (which does not exist anymore) is that I do not think that it does not make sense to log that nothing was inserted